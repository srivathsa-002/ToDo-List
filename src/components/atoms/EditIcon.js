import React from 'react';
import CreateOutlinedIcon from '@material-ui/icons/CreateOutlined';
import { Tooltip, IconButton } from "@material-ui/core";

export default class EditIcon extends React.Component {
    render() {
        return (
            <Tooltip data-testid="edit" onClick={this.props.click} title="Edit ToDo" placement={this.props.placement} arrow>
                <div>
                    <IconButton>
                        <CreateOutlinedIcon />
                    </IconButton>
                </div>
            </Tooltip>
        );
    }
}