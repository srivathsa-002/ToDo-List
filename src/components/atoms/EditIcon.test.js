import React from "react";
import {render,fireEvent} from "@testing-library/react";
import EditIcon from "./EditIcon";
import "@testing-library/jest-dom/extend-expect";
import {create} from "react-test-renderer";

describe("EditIcon testing", () => {
    test("Matches the snapshot", () => {
      const input = create(<EditIcon />);
      expect(input.toJSON()).toMatchSnapshot();
    });
});

it("renders correctly", () => {
    const{getByTestId} = render(<EditIcon />)

    expect(getByTestId("edit")).toBeInTheDocument()
    // expect(queryByPlaceholderText("Todo...")).toBeTruthy()
})

// describe("Input value", () => {
//     it("updates on change", () => {
//         const {queryByPlaceholderText} = render(<InputBox />)
//         const inputBox = queryByPlaceholderText("Todo...")
//         fireEvent.change(inputBox, {target: {value: "test"}})
//         expect(inputBox.value).toBe("test")
//     })
// })
