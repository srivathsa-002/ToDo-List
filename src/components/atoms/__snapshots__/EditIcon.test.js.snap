// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditIcon testing Matches the snapshot 1`] = `
<div
  aria-describedby={null}
  className=""
  data-testid="edit"
  onBlur={[Function]}
  onFocus={[Function]}
  onMouseLeave={[Function]}
  onMouseOver={[Function]}
  onTouchEnd={[Function]}
  onTouchStart={[Function]}
  title="Edit ToDo"
>
  <button
    className="MuiButtonBase-root MuiIconButton-root"
    disabled={false}
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiIconButton-label"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM5.92 19H5v-.92l9.06-9.06.92.92L5.92 19zM20.71 5.63l-2.34-2.34c-.2-.2-.45-.29-.71-.29s-.51.1-.7.29l-1.83 1.83 3.75 3.75 1.83-1.83c.39-.39.39-1.02 0-1.41z"
        />
      </svg>
    </span>
    <span
      className="MuiTouchRipple-root"
    />
  </button>
</div>
`;
